simpleGraph/localAlignment.go

```diff
@@ -120,9 +122,13 @@ func SmithWaterman(alpha []dna.Base, beta []dna.Base, scores [][]int64, gapPen i
 	return m[maxI][maxJ], route, minI, maxI, minJ, maxJ
 }
 
-func tripleMaxTrace(a int64, b int64, c int64) (int64, rune) {
+func tripleMaxTrace(prev int64, a int64, b int64, c int64) (int64, rune) {
```

Comment:
why do we need to pass in prev? i am having a hard time wrapping why head around that

Reply:
So the way tripleMaxTrace determines if it is a match or a mismatch is by determining if the score goes up (match) or down (mismatch). In order to determine how the score changed, I need to know what the score was from the previous iteration.